### PR TITLE
Change GeoSvc to accept vector of xmls

### PR DESCRIPTION
**Pull request by jlingema**
@clementhelsens wrote the detector descriptions in separate xml-files which allows for more flexibility. This PR does a small change to the `GeoSvc` to allow using multiple xml-files.

### CODE REVIEW

File: DetectorDescription/DetDesServices/src/GeoSvc.cxx [hunk 1]
```diff
@@ -51,9 +51,11 @@ StatusCode GeoSvc::buildDD4HepGeo(){
     m_dd4hepgeo->addExtension<IGeoSvc>(this);
 
     //load geometry
-    m_log << MSG::INFO << "loading geometry from file:  '" << m_xmlFileName << "'" << endmsg;
-    char* arg = (char*) m_xmlFileName.c_str();
-    m_dd4hepgeo->apply("DD4hepXMLLoader", 1, &arg);
+    for (auto& filename : m_xmlFileNames) {
+        m_log << MSG::INFO << "loading geometry from file:  '" << filename << "'" << endmsg;
+        char* arg = (char*) filename.c_str();
```
Comment (hegner) [inline DetectorDescription/DetDesServices/src/GeoSvc.cxx]: no c-style casts please!

File: DetectorDescription/Detectors/src/HCalBarrel_geo.cxx [hunk 2]
```diff
@@ -110,11 +108,11 @@ static DD4hep::Geometry::Ref_t createHCal (
         dxWedge1, dxWedge2, dzModule, dzModule, drWedge
       ), lcdd.material("Air")
   );
-  for (unsigned int idxLayer = 0; idxLayer < numModulesR; ++idxLayer) {
+  for (unsigned int idxLayer = 1; idxLayer < numModulesR + 1; ++idxLayer) {
```
Comment (hegner) [inline DetectorDescription/Detectors/src/HCalBarrel_geo.cxx]: the +1 may deserve a comment.

File: DetectorDescription/Detectors/src/HCalBarrel_geo.cxx [hunk 3]
```diff
@@ -110,11 +108,11 @@ static DD4hep::Geometry::Ref_t createHCal (
...
     auto layerName = std::string("wedge") + DD4hep::XML::_toString(idxLayer, "layer%d");
     unsigned int sequenceIdx = idxLayer % 2;
     double rminLayer = idxLayer * moduleDimensions.dr();
-    double rmaxLayer = (idxLayer + 1) * moduleDimensions.dr();
+    double rmaxLayer = idxLayer + 1 * moduleDimensions.dr();
```
Comment (hegner) [inline DetectorDescription/Detectors/src/HCalBarrel_geo.cxx]: the " 1 * " could be removed, right?

Comment (jlingema) [inline DetectorDescription/Detectors/src/HCalBarrel_geo.cxx]: Actually, the +1 is needed here. After having a closer look I realised that when I changed (back in December) the way I construct the components I introduced a bug: I calculate the width of the first layer of modules from the radial position and forgot to add the lower radius of the barrel HCal leading to 0-width modules in the first layer.

**Comment by hegner** [comment]
@jlingema - Could you fix your PR for the current conflicts?

**Comment by jlingema** [comment]
Fixed.

**Pull request merged**